### ISSUE TITLE
VlnPlot fails when feature and metadata share name

### ISSUE THREAD

**Issue by Alanocallaghan** [satijalab/seurat#2242]
If a metadata column and a feature share the same name, then VlnPlot fails. See reprex below:

``` r
library("Seurat")
#> Registered S3 method overwritten by 'R.oo':
#>   method        from       
#>   throw.default R.methodsS3
data(pbmc_small)

VlnPlot(pbmc_small, features="MS4A1")
```

![](https://i.imgur.com/FILgVOo.png)

``` r

pbmc_small$MS4A1 <- pbmc_small@assays$RNA@scale.data[1, ]


VlnPlot(pbmc_small, features="MS4A1")
#> Error in `[.data.frame`(data, , x, drop = FALSE): undefined columns selected
```

<sup>Created on 2019-10-23 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by Alanocallaghan** [comment]
This is because FetchData looks in both the metadata (https://github.com/satijalab/seurat/blob/9e4ed092cb2c3c0a883b642951cdead911bff9aa/R/objects.R#L931) and in the specified slot (https://github.com/satijalab/seurat/blob/9e4ed092cb2c3c0a883b642951cdead911bff9aa/R/objects.R#L934). 

I think it's a bit iffy for one function to just search through and pass back everything it finds with that name regardless of location (imagine a gene named PC1 for example).

**Comment by andrewwbutler** [comment]
Hi @Alanocallaghan ,

Thanks for the bug report and reprex! This should be resolved in the `develop` branch now. 

For cases like your example, the expected behavior would be to plot the values from meta.data column. If you are in the situation where you have a meta.data column name that is also a feature name, you can disambiguate by appending the assay key to the name (e.g. `"rna_MS4A1"`) and it should always pull the proper data.

**Comment by Alanocallaghan** [comment]
Cheers Andrew! That makes sense for the most part. I think in an ideal world it might be better to use separate functions for fetching metadata and for fetching assay data, but I realise that could be a pretty nasty refactor and also would definitely be a breaking change for users.

Fixed in 2877837f29668c224ca454f060cf610fa2c7615b
